### PR TITLE
Entity list/user preferences stored in Mongo

### PR DESCRIPTION
**Pull request by luk-kaminski**
## Description
Entity list/user preferences stored in Mongo.
BE part of #14293

## Motivation and Context
Replace temporary in-memory implementation with proper, Mongo-based implementation.
Limit the number of properties that FE really needs to store in a permanent storage.
/nocl

## How Has This Been Tested?
Manually and with additional tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/rest/resources/entities/preferences/EntityListPreferencesResource.java
Comment (todvora) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/entities/preferences/EntityListPreferencesResource.java:66]: Should we resolve this TODO before merging?

**Review by todvora — APPROVED** [review]
Looks good! Thanks!

File: graylog2-server/src/main/java/org/graylog2/rest/resources/entities/preferences/EntityListPreferencesResource.java
Comment (luk-kaminski) [reply graylog2-server/src/main/java/org/graylog2/rest/resources/entities/preferences/EntityListPreferencesResource.java:66]: _TLDR - I do not see any big risk, but we can wait till the evening to merge after release, to be on the safe side._

I personally think that we should not store audit log every time a user changes visible columns or page size in any entity list. I'd love to consult it with Kay (he is back in a few days). If he had different opinion, I'll simply add it.
For now nobody uses this endpoints, there is no FE ready. 
It would be nice to have it merged so that FE works on the final solution when developing FE part.
I suggest merging after release and quickly adding audit logging if needed.

**Pull request merged**
